### ISSUE TITLE
`wildcard_imports` (with `warn-on-all-wildcard-imports`) warns about wildcards within external macros

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust-clippy#15412]
### Summary

Not much more to say, the lint fires within external macros where I can't do anything about their usage.

### Lint Name

wildcard-imports

### Reproducer

I tried this code:

```rust
#![warn(clippy::wildcard_imports)]

fn main() {
    let tokens = Vec::<proc_macro2::TokenTree>::new();
    dbg!(quote::quote!(#(#tokens)*));
}
```

with a `clippy.toml`:

```toml
warn-on-all-wildcard-imports = true
```

I saw this happen:

```
warning: usage of wildcard import
 --> src/main.rs:5:10
  |
5 |     dbg!(quote::quote!(#(#tokens)*));
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_imports
```

I expected to see this happen: No warning.

### Version

```text
rustc 1.90.0 (b56aaec52 2025-07-24)
binary: rustc
commit-hash: b56aaec52bc0fa35591a872fb4aac81f606e265c
commit-date: 2025-07-24
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```
